iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/IcebergTableUtil.java

```diff
@@ -184,4 +186,27 @@ public static void updateSpec(Configuration configuration, Table table) {
   public static boolean isBucketed(Table table) {
     return table.spec().fields().stream().anyMatch(f -> f.transform().toString().startsWith("bucket["));
   }
+
+  /**
+   * Roll an iceberg table's data back to a specific snapshot identified either by id or before a given timestamp.
+   * @param configuration a Hadoop configuration
+   * @param table the iceberg table
+   */
+  public static void rollback(Configuration configuration, Table table) {
+    RollbackSpec rollbackSpec = SessionStateUtil.getResource(configuration, hive_metastoreConstants.ROLLBACK_SPEC)
```

Comment:
I refactored the code to avoid using the SessionState to pass execute operation parameters. From now every info is passed through the `AlterTableExecuteSpec` and the storagehandler will do the rollback instead of the metahook.